components/text_recognition/browser/text_recognition.mm

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/text_recognition/browser/text_recognition.h"
+
+#import <Foundation/Foundation.h>
+#import <Vision/Vision.h>
+
+#include "base/logging.h"
+#include "base/mac/foundation_util.h"
+#include "base/mac/mac_util.h"
+#include "base/mac/scoped_cftyperef.h"
+#include "base/strings/sys_string_conversions.h"
+#include "skia/ext/skia_utils_base.h"
+#include "skia/ext/skia_utils_mac.h"
+#include "third_party/skia/include/core/SkBitmap.h"
+#include "third_party/skia/include/utils/mac/SkCGUtils.h"
+
+std::vector<std::string> GetTextFromImage(const SkBitmap& image) {
+  std::vector<std::string> result;
+  if (@available(macOS 10.15, *)) {
+    // The bitmap type is sanitized to be N32 before we get here. The conversion
+    // to an NSImage would not explode if we got this wrong, so this is not a
+    // security CHECK.
+    DCHECK_EQ(image.colorType(), kN32_SkColorType);
+
+    auto* p_result = &result;
+    VNRecognizeTextRequest* textRecognitionRequest =
+        [[VNRecognizeTextRequest alloc] initWithCompletionHandler:^(
+                                            VNRequest* _Nonnull request,
+                                            NSError* _Nullable error) {
+          NSArray<VNRecognizedTextObservation*>* observations = request.results;
+
+          [observations enumerateObjectsUsingBlock:^(
+                            VNRecognizedTextObservation* _Nonnull obj,
+                            NSUInteger idx, BOOL* _Nonnull stop) {
+            // Ask first top candidate for a recognized text string.
+            VNRecognizedText* recognizedText =
+                [obj topCandidates:1].firstObject;
+
+            p_result->push_back(
+                base::SysNSStringToUTF8([recognizedText string]));
+          }];
+        }];
+
+    textRecognitionRequest.recognitionLevel =
+        VNRequestTextRecognitionLevelAccurate;
+    textRecognitionRequest.usesLanguageCorrection = true;
+    // Copied FF's supported language list.
+    // See https://support.mozilla.org/en-US/kb/text-recognition
+    // English, Chinese, Portuguese, French, Italian, German, and Spanish
+    textRecognitionRequest.recognitionLanguages = @[
+      @"en-US", @"zh-Hans", @"zh-Hant", @"pt-BR", @"fr-FR", @"it-IT", @"de-DE",
```

Comment:
> The order of the languages in the array defines the order in which languages are used during language processing and text recognition.

I guess it could be cool to get the webpage language and prioritize it here, but I'm okay with the current approach :)